pkgs/by-name/st/stackblur-go/package.nix

```
@@ -0,0 +1,32 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+}:
+
+buildGoModule rec {
+  pname = "stackblur-go";
+  version = "1.1.0";
+
+  src = fetchFromGitHub {
+    owner = "esimov";
+    repo = "stackblur-go";
+    rev = "v${version}";
+    hash = "sha256-y1Fov81mholhz+bLRYl+G7jhzcsFS5TUjQ3SUntD8E0=";
+  };
+
+  vendorHash = null;
+
+  postInstall = ''
+    mv $out/bin/cmd $out/bin/stackblur
+  '';
+
+  ldflags = [ "-s" "-w" ];
+
+  meta = with lib; {
+    description = "A fast, almost Gaussian Blur implementation in Go";
+    homepage = "https://github.com/esimov/stackblur-go";
+    license = licenses.mit;
+    maintainers = with maintainers; [ sodiboo ];
+    mainProgram = "stackblur";
+  };
```

```suggestion
  meta = {
    description = "A fast, almost Gaussian Blur implementation in Go";
    homepage = "https://github.com/esimov/stackblur-go";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ sodiboo ];
    mainProgram = "stackblur";
  };
```

The use of `with lib;` is discouraged. ref: https://github.com/NixOS/nixpkgs/pull/293767
You can put `lib.licenses.mit` in `meta.license`, and others too.

---

Hm, i was vaguely aware of this being discouraged, but i had understood it as though nixpkgs encourages `meta = with lib;`, since i've seen a lot of packages do exactly that, including some i've worked on [[1]](https://github.com/NixOS/nixpkgs/blob/731666d8d7b536cb09deec240dfba3fa1c058ca2/pkgs/by-name/ni/niri/package.nix#L80) [[2]](https://github.com/NixOS/nixpkgs/blob/731666d8d7b536cb09deec240dfba3fa1c058ca2/pkgs/games/simutrans/default.nix#L155).

In this case, i feel it's probably fine? it's obvious enough where `licenses` and `maintainers` comes from and there are literally no other identifiers within its scope (except `sodiboo` but that one is even more obvious). I could change it i guess, but it just feels like changing it for the sake of changing it?

---

Thanks for your reply. I understand your opinion, `meta` in stackblur-go certainly only contains `lib` identifiers. However, I think `meta` can be extended to include other fields that can access other identifiers within its scope (e.g. `meta.broken` and `meta.homepage`, etc.), on the other hand `meta.maintainers` hardly contains them. Considering that other properties will be added in the future, I think using `lib.licenses` without `with lib;` contributes to long-term maintainability.
Although, to be honest, I have only recently read 293767 and started reviewing nixpkgs, so I may be misreading it, I feel the point that using `with lib;` for `meta` is too broad is a legitimate one.

Of course, more discussion is needed on whether these changes help the project, and I welcome your opinions.

---

I've implemented the suggested changes in a recent commit.